pkg/pipeline/BulkUpdateService.go

```diff
@@ -0,0 +1,302 @@
+		Script:    script,
+		ReadMe:    bulkUpdateReadme.Readme,
+	}
+	return response, nil
+}
+func (impl BulkUpdateServiceImpl) GetBulkAppName(bulkUpdatePayload BulkUpdatePayload) ([]*ImpactedObjectsResponse, error) {
+	impactedObjectsResponse := []*ImpactedObjectsResponse{}
+	if len(bulkUpdatePayload.Includes.Names) == 0 {
+		return impactedObjectsResponse, nil
+	}
+	if bulkUpdatePayload.Global {
+		appsGlobal, err := impl.bulkUpdateRepository.
+			FindBulkAppNameForGlobal(bulkUpdatePayload.Includes.Names, bulkUpdatePayload.Excludes.Names)
+		if err != nil {
+			impl.logger.Errorw("error in fetching bulk app names for global", "err", err)
+			return nil, err
+		}
+		for _, app := range appsGlobal {
+			impactedObject := &ImpactedObjectsResponse{
+				AppId:   app.Id,
+				AppName: app.AppName,
+			}
+			impactedObjectsResponse = append(impactedObjectsResponse, impactedObject)
+		}
+	}
+	for _, envId := range bulkUpdatePayload.EnvIds {
+		appsNotGlobal, err := impl.bulkUpdateRepository.
+			FindBulkAppNameForEnv(bulkUpdatePayload.Includes.Names, bulkUpdatePayload.Excludes.Names, envId)
+		if err != nil {
+			impl.logger.Errorw("error in fetching bulk app names for env", "err", err)
+			return nil, err
+		}
+		for _, app := range appsNotGlobal {
+			impactedObject := &ImpactedObjectsResponse{
+				AppId:   app.Id,
+				AppName: app.AppName,
+				EnvId:   envId,
+			}
+			impactedObjectsResponse = append(impactedObjectsResponse, impactedObject)
+		}
+	}
+	return impactedObjectsResponse, nil
+}
+func (impl BulkUpdateServiceImpl) ApplyJsonPatch(patch jsonpatch.Patch, target string) (string, error) {
+	modified, err := patch.Apply([]byte(target))
+	if err != nil {
+		impl.logger.Error("error in applying JSON patch")
+		return "Patch Failed", err
+	}
+	return string(modified), err
+}
+func (impl BulkUpdateServiceImpl) BulkUpdateDeploymentTemplate(bulkUpdatePayload BulkUpdatePayload) BulkUpdateResponse {
+	var bulkUpdateResponse BulkUpdateResponse
+	if len(bulkUpdatePayload.Includes.Names) == 0 {
+		bulkUpdateResponse.Message = append(bulkUpdateResponse.Message, "Please don't leave includes.names array empty")
+		return bulkUpdateResponse
+	}
+	patchJson := []byte(bulkUpdatePayload.DeploymentTemplate.Spec.PatchJson)
+	patch, err := jsonpatch.DecodePatch(patchJson)
+	if err != nil {
+		impl.logger.Errorw("error in decoding JSON patch", "err", err)
+		bulkUpdateResponse.Message = append(bulkUpdateResponse.Message, "The patch string you entered seems wrong, please check and try again")
+		return bulkUpdateResponse
+	}
+	var charts []*chartConfig.Chart
+	if bulkUpdatePayload.Global {
+		charts, err = impl.bulkUpdateRepository.FindBulkChartsByAppNameSubstring(bulkUpdatePayload.Includes.Names, bulkUpdatePayload.Excludes.Names)
+		if err != nil {
+			impl.logger.Error("error in fetching charts by app name substring")
+			bulkUpdateResponse.Message = append(bulkUpdateResponse.Message, "Unable to bulk update apps globally : error in connecting with db")
```

Comment:
we shoul use err for response as this may not be db connection error

Reply:
also this cant be the message at the top most level, top most level message should be summary of what happened... also this ll be overwrriten by error coming later